oaknational/Oak-Web-Application
# Feat: don't show 'your details'  for fully authenticated users LESQ-1003

## Description

When feature flag enabled

- If a user if fully onboarded don't show the 'Your details' section on the downloads and share pages
- During onboarding update local storage to match details submitted to hubspot (terms, email, school)

For a logged in fully onboarded user
- When download and share page loads fetch school and subscription status from hubspot
- Update local storage and downloads form with fetched data
- Don't show details on page  
- Download and copyright button under checkboxes

For a logged on user without isOnboarded in meta data 
- When download and share page loads fetch school and subscription status from hubspot
- Update local storage and downloads form with fetched data
- Show details on page - pre populated with hubspot data

Downloads and share should work the same as before if not logged in

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2828--oak-web-application.netlify.thenational.academy

**auth feature flag disabled** - downloads and share work as expected

**onboarding**
- Local storage is updated with onboarding form details (terms, school, email (if newsletter subscription is accepted)) nb you wont see the newsletter checkbox if you have already accepted this with email.
- Hubspot receives correct data

**downloads and share fully onboarded**
- fully onboarded teacher users should not see 'your details'.
- button moved as [figma](https://www.figma.com/design/fxWHjjNzvbbc0J38pM3Kti/Log-in?node-id=3094-22115&t=Bett0Rr9vdIadGxb-1) design.
- correct data is stored in local storage on download 
- correct data is sent to hubspot on downloads 

**downloads and share partially onboarded**
- No teacher users should see 'your details'.
- Details should be pre-populated with correct data
- correct data is stored in local storage on download 
- correct data is sent to hubspot on downloads 



## Checklist
